src/api/data.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * This module is private and deal with updating a grpah object internal Immutable.Map
+ */
+import curry from 'lodash/curry';
+import isString from 'lodash/isString';
+import Immutable from 'immutable';
+
+/**
+ * return true if the parameter is an Immutable.Map throw otherwise
+ * @param {any} map
+ * @return {Bool}
+ */
+export function isMapElseThrow(map) {
+	const test = Immutable.Map.isMap(map);
+	if (!test) {
+		throw new Error(
+			`can't process value as the target is not an Immutable.Map was given a ${map &&
+				map.toString()}.`,
+		);
+	}
+	return test;
+}
+
+/**
+ * return true if the parameter key is a String throw otherwise
+ * @param {any} map
+ * @return {Bool}
+ */
+export function isKeyElseThrow(key) {
+	const test = isString(key);
```

Comment:
from what I know Immutable.map can deal with integer key,
and others types also even if not recommended

Reply:
this is totaly intended to force user to rely on a something more reliable than the js number type (that struggle to represent accurately high number)